#2911: DS Data Redesign [AD]

dave-kennedy-ecs opened the pull request:
## Ticket

<!-- PR title format: `#issue_number: Descriptive name ideally matching ticket name - [sandbox]`-->
Resolves #2911 
Resolves #1242 
Resolves #3680 

## Changes

<!-- What was added, updated, or removed in this PR. -->
- New DS Data form design
- Cancel modal added to nameservers form
- DS Data Form details:
  - New empty state for DS data with new Add DS Data button and removed info message
  - Add new DS data form with Save / cancel when no DS data exists and Add DS Data button pressed
  - DS data table when DS data exists, table containing readonly rows with Edit button, and Delete button in a kebab; when edit pressed, edit row contains editable DS data, and action buttons for Save, Cancel and Delete
  - Add new DS data form with Save / cancel appears above DS data table when Add DS Data button pressed
  - Modal appears when unsaved changes and certain actions such as Edit and Add DS Data button pressed and form changed
  - Modal appears when unsaved changes and cancel button is clicked
  - Modal appears when delete button is pressed, with a variation in the message when attempting to delete the last ds data record
  - Fix ALL table headers on tables in mobile view

<!--
    Please add/remove/edit any of the template below to fit the needs
    of this specific PR.
--->

## Context for reviewers

<!--Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.  -->

## Setup

<!--  Add any steps or code to run in this section to help others run your code.
    
    Example 1:
    ```sh
    echo "Code goes here"
    ``` 
    
    Example 2: If the PR was to add a new link with a redirect, this section could simply be:
    -go to /path/to/start/page
    -click the blue link in the <insert location>
    -notice user is redirected to <proper end location>
-->

## Code Review Verification Steps

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)
<!-- Mark "- N/A" and check at the end of each check that is not applicable to your PR -->
- [x] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### As a designer reviewer, I have

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

## Screenshots

<!-- If this PR makes visible interface changes, an image of the finished interface can help reviewers
and casual observers understand the context of the changes.
A before image is optional and can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow.
You may want to use [GIPHY Capture](https://giphy.com/apps/giphycapture) for this! 📸

_Please frame images to show useful context but also highlight the affected regions._
--->

gabo0oo commented:
Hello! Posting my complete review below in separate messages.

gabo0oo commented:
Body text: Change hatch marks to typographers quotes in Add DS data in second paragraph.

gabo0oo commented:
- “Required fields are marked...” line should appear below the <p> content under above the Add DS record card.
  - This has also reverted on the name servers card but it’s also supposed to be on the outside.
- Add DS record card should have these padding numbers (this needs to be updated on the name servers card as well)
  - left, right, bottom: 24
  - top: 16
- I think I found a bug (see screenshots). I entered some mock data into the Add DS record card, then canceled. When I went to add the DS record again, the Digest field was showing the red error outline as if it triggered an error, but I hadn’t entered anything yet. Additionally, when I clicked ‘Save’ with no data entered in any of the fields, it successfully saved my record.
<img width="612" alt="Screenshot 2025-03-18 at 3 20 00 PM" src="https://github.com/user-attachments/assets/d93cde30-83df-4b3d-ae51-43347ac0e784" />
<img width="622" alt="Screenshot 2025-03-18 at 3 20 13 PM" src="https://github.com/user-attachments/assets/51411090-35df-4adc-89b8-c37142a0965a" />

gabo0oo commented:
- DS record table (Looking at 1024 breakpoint) Can we make key tag and Digest type fields closer to the same widths. I’m noticing that when I click edit on the row, the key tag column widens a bit and the column header goes to 1 line. I’m wondering if we could get Digest type to sit at 1 line as well. Looks like key tag is around 91px wide. I think we could make Digest type just a little bit wider than this to get it to go to one line. This would mean borrowing some of the space from the Digest column which I think is ok at this smaller viewport size.
  - Can we make ths and tds have a 16px right padding instead of 32? I think this would help with the squeezing that’s happening on the columns.
- DS record table potential bug:
  - Entered in ec0bdd990b39feead889f0ba613db4adec0bdd99qwertyuasdfghjkzxcvbnmqw for SHA 256 but keep getting the errors:
    - Digest must contain only alphanumeric characters (0-9, a-f).
    - SHA-256 digest must be exactly 64 characters.
    - Note in this case I just randomly typed in letters and numbers. When I actually generated some digest data and input it I was able to successfully save
  - That entry is 64 characters and just uses alphanumeric characters.
- Are you sure you want to continue Modal:
  - The margins on this modal seem off. There’s an extra margin of 44.5 that I’m seeing. This isn’t present on the Cancel modal.

gabo0oo commented:
- DS record table: Can we adjust the kebab dropdown a bit to match our other dropdowns (I missed this on the name servers update so it needs to be changed there).
  - Space between text button and ‘More options’ should be 16 not 24
  - kebab focus state seems like its missing a little bit of bottom padding, like the background color does quite reach low enough.
  - Text button should be underlined to match our other dropdowns.

example from production of our dropdown for remove member
<img width="202" alt="Screenshot 2025-03-18 at 3 57 13 PM" src="https://github.com/user-attachments/assets/ef166158-fb3f-4faa-a533-414625620ca5" />

gabo0oo commented:
- DS record table error states: I’m seeing the error icons behaving funny, some of them are sizing very small.  Icons should be the smaller 16px size.
- Error message text should use the xs small bold type style

gabo0oo commented:
Ok, that's all!

dave-kennedy-ecs commented:
> Body text: Change hatch marks to typographers quotes in Add DS data in second paragraph.

done

dave-kennedy-ecs commented:
> “Required fields are marked...” line should appear below the
> content under above the Add DS record card.
> 
> * This has also reverted on the name servers card but it’s also supposed to be on the outside.

Fixed this on both DS Data and Nameservers

dave-kennedy-ecs commented:
> * I think I found a bug (see screenshots). I entered some mock data into the Add DS record card, then canceled. When I went to add the DS record again, the Digest field was showing the red error outline as if it triggered an error, but I hadn’t entered anything yet. 
> 
> <img alt="Screenshot 2025-03-18 at 3 20 00 PM" width="612" src="https://private-user-images.githubusercontent.com/107440934/424231713-d93cde30-83df-4b3d-ae51-43347ac0e784.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzODMyMjYsIm5iZiI6MTc0MjM4MjkyNiwicGF0aCI6Ii8xMDc0NDA5MzQvNDI0MjMxNzEzLWQ5M2NkZTMwLTgzZGYtNGIzZC1hZTUxLTQzMzQ3YWMwZTc4NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxOVQxMTE1MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZTIwNmRiMjZiMjM1NGUzZDU3MmU3NjExOWI2Nzg2ZjNlNzI3NmM1NGRjYjE5MDQyYzYxY2FkMDQ0ZDAzNjdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.V8NVCHdGqhPO18kVcPvD2bvTrNk9RGffqW8GKhkOAjM"> 

This bug is now fixed.

dave-kennedy-ecs commented:
> * Additionally, when I clicked ‘Save’ with no data entered in any of the fields, it successfully saved my record.
> 
> <img alt="Screenshot 2025-03-18 at 3 20 13 PM" width="622" src="https://private-user-images.githubusercontent.com/107440934/424231721-51411090-35df-4adc-89b8-c37142a0965a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDIzODMyMjYsIm5iZiI6MTc0MjM4MjkyNiwicGF0aCI6Ii8xMDc0NDA5MzQvNDI0MjMxNzIxLTUxNDExMDkwLTM1ZGYtNGFkYy04OWI4LWMzNzE0MmEwOTY1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxOVQxMTE1MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZmQ1ZmMxMjBkNDZmMjBmY2ZkMmI4OTFhNjM5ODZiMzM5ODY1MGIyNDE3MDJmMzY1MzhhZDIyNzIzNjNhZWVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lEgAm0KU760n5LYn5wM6iQjg536BBtmu2Fofj_76sl8">

Can you provide some guidance as to what the expected behavior is when the initial add card is submitted without being filled out?

dave-kennedy-ecs commented on src/registrar/templates/domain_nameservers.html:
```diff
@@ -82,7 +82,7 @@ <h1 class="tablet:margin-bottom-1">Name servers</h1>
 
-          <section class="add-nameservers-form display-none section-outlined">
+          <section class="add-nameservers-form display-none section-outlined section-outlined--extra-padding">
             {{ form.domain }}
```
> **Blocking accessibility items on the "add ds data" modal:**
> 
> 1. (minor cleanup item) digest is referencing two ids in aria-describedby. One of these ids do not exist, I think this is just old content. Reads the same.
>    ![image](https://private-user-images.githubusercontent.com/141044360/425062984-0f880c8e-778a-48e2-926e-e49b1240f374.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI0OTkyMTIsIm5iZiI6MTc0MjQ5ODkxMiwicGF0aCI6Ii8xNDEwNDQzNjAvNDI1MDYyOTg0LTBmODgwYzhlLTc3OGEtNDhlMi05MjZlLWU0OWIxMjQwZjM3NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyMFQxOTI4MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YzgzZDMzYzBiODMxNTAwNGJhZDVhZDYxMzg0NTRhYjhkODhiZDI5NzNmMTIxOGRmZTMzZDAzY2U0ZjhjNDQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VRM2kyuMVHS1sOBZa8FvYedXGZqpSOMXMd5RS9lc_go)

This is fixed

> 2. The "Save" button needs an aria-label

None of the other Save buttons throughout the app have aria-labels. Is it necessary here?

> 3. When clicking the add ds data button, this change is not announced to screenreaders. Thus, these users do not know that a change occured. You can use the `announce()` function in our helpers.

What should be announced? The visibility of a form?

dave-kennedy-ecs replied:
CC: @SamiyahKey please see Zander's feedback on feasibility and advise if we need to take any action on it.

dave-kennedy-ecs commented on src/registrar/assets/src/js/getgov/form-dsdata.js:
```diff
@@ -0,0 +1,519 @@
+import { showElement, hideElement, scrollToElement } from './helpers';
+import { removeErrorsFromElement, removeFormErrors } from './form-helpers';
+
+export class DSDataForm {
```
I did not initially inherit because I did not know how different or similar nameserver and ds data would be in implementation. I wanted to complete separate implementations before determining inheritance. There is a benefit to inheritance here.

rachidatecs commented on src/registrar/assets/src/js/getgov/form-dsdata.js:
```diff
@@ -0,0 +1,519 @@
+        // Set event listeners on the submit buttons for the modals. Event listeners
+        // should execute the callback function, which has its logic updated prior
+        // to modal display
+        const unsaved_changes_modal = document.getElementById('unsaved-changes-modal');
+        if (unsaved_changes_modal) {
+            const submitButton = document.getElementById('unsaved-changes-click-button');
+            const closeButton = unsaved_changes_modal.querySelector('.usa-modal__close');
+            submitButton.addEventListener('click', () => {
+                closeButton.click();
+                this.executeCallback();
+            });
+        }
+        const cancel_changes_modal = document.getElementById('cancel-changes-modal');
```
cc @gabo0oo 

This question came up when we implemented a similar modal on domain management page. We determined that it would be expensive to intercept all link clicks (and we can't use pushState like in a react app) so we left the nav links alone - they do not trigger the modal.

rachidatecs commented on src/registrar/templates/domain_nameservers.html:
```diff
@@ -82,7 +82,7 @@ <h1 class="tablet:margin-bottom-1">Name servers</h1>
           {% endcomment %}
 
-          <section class="add-nameservers-form display-none section-outlined">
+          <section class="add-nameservers-form display-none section-outlined section-outlined--extra-padding">
```
> **Blocking accessibility items on this table**:
> 
> 1. On mobile view, the table does not show the headers. This is because the each < td > is missing a `data-label=''` tag. USWDS also uses this for other accessibility items so this should be added

I created a [ticket](https://github.com/cisagov/manage.get.gov/issues/3680) to handle these misses on all tables. Having said that, I think Dave already fixed it on this one. I'm just gonna pull the ticket and fixed it on the other tables and have you review those too, to thank you for your efforts.

rachidatecs replied:
Done.

dave-kennedy-ecs commented:
> (**BLOCKING DUE TO #1242**) Found an edge case with either the messaging or the key tag input: you can enter negative values. I think we do have an accessibility rule where we prefer to show errors rather than deny the input (as is the case with allow values greater than 65k) - so maybe alternatively the error message should be slightly adjusted here in this case? Will defer to design on this one, but either would work ![image](https://private-user-images.githubusercontent.com/141044360/425046873-0a228e23-e8c8-446e-a7e1-fa0ece6a4121.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MDE5OTQsIm5iZiI6MTc0MjUwMTY5NCwicGF0aCI6Ii8xNDEwNDQzNjAvNDI1MDQ2ODczLTBhMjI4ZTIzLWU4YzgtNDQ2ZS1hN2UxLWZhMGVjZTZhNDEyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyMFQyMDE0NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NGQ4MWI2N2E0ZTU1ZjI1NWM1ZTE2NjFiMTA3MGEzMGNlMDM5ZTQ0MTllY2YwZDg2NzhhOWYxZDUyNTg4NWMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5HUv0h6DOmJ78DXhP73zJzIkaVV2Krk0p1DYkl2XW4Y)

Updated error message to:
Key tag must be greater than 0 and less than 65535.

CC: @gabo0oo @SamiyahKey

gabo0oo commented:
The top and bottom padding on the rows is really small. Can we change it back to matching the name servers pattern we already have?
<img width="764" alt="Screenshot 2025-03-20 at 4 47 11 PM" src="https://github.com/user-attachments/assets/014069a9-85a5-4381-bc28-de91df6ca219" />
<img width="747" alt="Screenshot 2025-03-20 at 4 47 01 PM" src="https://github.com/user-attachments/assets/02bb5ce5-ed6b-49f4-b5bd-9511d60eacbb" />

gabo0oo commented:
I think we've gone too small on the right side padding of each td. I think 8 is too small. If the th get pushed to 2 lines because of this it's ok.

gabo0oo commented:
In addition to the odd ths missing on the data on mobile, I'm seeing a th repeat itself twice.
<img width="445" alt="Screenshot 2025-03-20 at 4 48 05 PM" src="https://github.com/user-attachments/assets/1e6099e5-72a0-4f15-b351-24213c623ccf" />

dave-kennedy-ecs commented:
> In addition to the odd ths missing on the data on mobile, I'm seeing a th repeat itself twice. <img alt="Screenshot 2025-03-20 at 4 48 05 PM" width="445" src="https://private-user-images.githubusercontent.com/107440934/425222451-1e6099e5-72a0-4f15-b351-24213c623ccf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NDgxNzksIm5iZiI6MTc0MjU0Nzg3OSwicGF0aCI6Ii8xMDc0NDA5MzQvNDI1MjIyNDUxLTFlNjA5OWU1LTcyYTAtNGYxNS1iMzUxLTI0MjEzYzYyM2NjZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyMVQwOTA0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MGRhY2ZhYWQyZDFmMjMyZGI3ZTYyYTNhMDllYTA1NzgxYWJlMTk4OTRjMTM2MWU1NmI3MWMxZGZiN2Q5YWIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rx06VjGr5R4BRTSDykZM49oLHtbFTGLF6IspCz669fI">

Fixed

dave-kennedy-ecs commented:
> 2. When the key tag is 0, the add form lets you save it. However, when clicking edit, the input field is cleared and upon save it throws an error on that field
> 
> You can view both of these in this loom here: https://www.loom.com/share/8d9954639c21411dbcd9bdbe62224dea
> 
> There is also a third bug not in the loom above: 3. The form does not throw an error when submitting duplicate records, it "fails silently" as with #1

All of these are now fixed. For the loom, data-initial-value was defaulting to blank for the value of 0 because 0 was testing as False. I changed the test to test for not None, and fixed data-initial-value. Duplicate checking is now in place as well.

gabo0oo commented on src/registrar/utility/errors.py:
```diff
@@ -260,7 +262,8 @@ class DsDataError(Exception):
         DsDataErrorCodes.INVALID_DIGEST_SHA256: ("SHA-256 digest must be exactly 64 characters."),
         DsDataErrorCodes.INVALID_DIGEST_CHARS: ("Digest must contain only alphanumeric characters (0-9, a-f)."),
-        DsDataErrorCodes.INVALID_KEYTAG_SIZE: ("Key tag must be less than 65535."),
+        DsDataErrorCodes.INVALID_KEYTAG_SIZE: ("Key tag must be greater than 0 and less than 65535."),
```
@SamiyahKey can I get your thoughts on this? The helper text for the Key tag field is "Numbers (0-9) only."  But the requirements are 1–65535. I'm not sure how we should adjust this error message and helper text. Perhaps:

Helper: "(1–65534)"
Error: "Enter a number between 1 and 65534"

dave-kennedy-ecs replied:
Sorry if I have misled here. The valid values are 0-65535. So positive and less than 65536

gabo0oo replied:
Change helper text to: 

> Numbers (0–65535) only


The error message looks good.

gabo0oo replied:
Following up on the above @rachidatecs @dave-kennedy-ecs

gabo0oo commented:
Since there is a 8 max we should add this line to the end of the 2nd body paragraph:

You can add a maximum of 13 DS records.

dave-kennedy-ecs commented:
> Since there is a 8 max we should add this line to the end of the 2nd body paragraph:
> 
> You can add a maximum of 13 DS records.

Done. Added content:
You can add a maximum of 8 DS records.

gabo0oo commented:
> I think we've gone too small on the right side padding of each td. I think 8 is too small. If the th get pushed to 2 lines because of this it's ok.

I'm still seeing the padding really small.

rachidatecs commented:
> > I think we've gone too small on the right side padding of each td. I think 8 is too small. If the th get pushed to 2 lines because of this it's ok.
> 
> I'm still seeing the padding really small.

I'd missed this. Fixed.

dave-kennedy-ecs commented:
2 additional comments. 
  I found this morning that migrations had to be run again on the sandbox, as this was causing EPP failures due to another branch being merged in without running migrations.
  And I also found and fixed an additional bug related to editing an existing ds data record when 8 ds data records already existed.  

At this point, I am not aware of any remaining issues that are not addressed, or any additional bugs that have been identified on this branch.
